### PR TITLE
Do not run scripts if there is a global top bin conflict

### PR DESCRIPTION
**Pull request by isaacs**
Do not run scripts if there is a global top bin conflict

Fix: https://github.com/github/npm-cli/issues/14

### CODE REVIEW

**Comment by isaacs** [comment]
CI failure due to macOS coveralls push timing out.  Addressed on different branch.

File: lib/arborist/rebuild.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const _linkAllBins = Symbol('linkAllBins')
 const _runScripts = Symbol('runScripts')
 const _buildQueues = Symbol('buildQueues')
 const _addToBuildSet = Symbol('addToBuildSet')
-const _checkBins = Symbol('checkBins')
+const _checkBins = Symbol.for('checkBins')
```
Comment (ljharb) [inline lib/arborist/rebuild.js:25]: why use Symbol.for instead of exporting a symbol that can be used explicitly?

**Review by darcyclarke — APPROVED** [review]
✅ LGTM

**Review by nlf — APPROVED** [review]
looks to get the job done :+1:

File: lib/arborist/rebuild.js (hunk 1 above)
Comment (isaacs) [reply lib/arborist/rebuild.js:25]: No compelling reason, really.  We use `Symbol.for()` in a few other places as well, when logic has to be shared between mixins.  Would be fine for a later refactor.

**Pull request closed without merging**